Modules/Core/Common/include/itkVector.hxx

```diff
@@ -155,7 +155,8 @@ template <typename T, unsigned int TVectorDimension>
 void
 Vector<T, TVectorDimension>::SetVnlVector(const vnl_vector<T> & v)
 {
-  for (unsigned int i = 0; i < v.size(); ++i)
+  const unsigned int elements_to_copy = std::min<unsigned int>(TVectorDimension, v.size());
+  for (unsigned int i = 0; i < elements_to_copy; ++i)
   {
     (*this)[i] = v(i);
   }
```

Comment:
I see, originally there would be undefined behavior if `v` would have more elements than `*this`. This adjustment should indeed prevent such undefined behavior. But another option might be to throw an exception, if `v.size() >  TVectorDimension`. Might be worth considering, I don't know 🤷

Reply:
@N-Dekker I considered your option as well.  I updated the documentation in the header file to indicate this behavior.   I thought this was the most similar to the old behavior.